cluster-autoscaler/cloudprovider/oci/nodepools/oci_manager.go

```diff
@@ -0,0 +1,678 @@
+		Name:   nodeName,
+		Labels: map[string]string{},
+	}
+
+	// Add all the initial node labels from the NodePool configuration to the
+	// templated node.
+	for _, kv := range nodePool.InitialNodeLabels {
+		node.ObjectMeta.Labels[*kv.Key] = *kv.Value
+	}
+
+	node.Status = apiv1.NodeStatus{
+		Capacity: apiv1.ResourceList{},
+	}
+
+	shape, err := m.ociShapeGetter.GetNodePoolShape(nodePool)
+	if err != nil {
+		return nil, err
+	}
+
+	taints, err := m.registeredTaintsGetter.Get(nodePool)
+	if err != nil {
+		klog.Warningf("could not extract taints from the nodepool: %s. Continuing on with empty taint list", err)
+		taints = []apiv1.Taint{}
+	}
+	node.Spec = apiv1.NodeSpec{
+		Taints: taints,
+	}
+	if shape.GPU > 0 {
+		node.Spec.Taints = append(node.Spec.Taints, apiv1.Taint{
+			Key:    "nvidia.com/gpu",
+			Value:  "",
+			Effect: "NoSchedule",
+		})
+	}
+
+	node.Status.Capacity[apiv1.ResourcePods] = *resource.NewQuantity(110, resource.DecimalSI)
+
+	node.Status.Capacity[apiv1.ResourceCPU] = *resource.NewQuantity(int64(shape.CPU), resource.DecimalSI)
+	node.Status.Capacity[apiv1.ResourceMemory] = *resource.NewQuantity(int64(shape.MemoryInBytes), resource.DecimalSI)
+	node.Status.Capacity[ipconsts.ResourceGPU] = *resource.NewQuantity(int64(shape.GPU), resource.DecimalSI)
+
+	node.Status.Allocatable = node.Status.Capacity
+
+	availabilityDomain, err := getNodePoolAvailabilityDomain(nodePool)
+	if err != nil {
+		return nil, err
+	}
+
+	node.Labels = cloudprovider.JoinStringMaps(node.Labels, buildGenericLabels(nodePool, nodeName, availabilityDomain))
+
+	node.Status.Conditions = cloudprovider.BuildReadyConditions()
+	return &node, nil
+}
+
+// getNodePoolAvailabilityDomain determines the availability of the node pool.
+// This breaks down if the customer specifies more than one placement configuration,
+// so best practices should be a node pool per AD if customers care about it during scheduling.
+// if there are more than 1AD defined, then we return the first one always.
+func getNodePoolAvailabilityDomain(np *oke.NodePool) (string, error) {
+	if len(np.NodeConfigDetails.PlacementConfigs) == 0 {
+		return "", fmt.Errorf("node pool %q has no placement configurations", *np.Id)
+	}
+
+	if len(np.NodeConfigDetails.PlacementConfigs) > 1 {
+		klog.Warningf("node pool %q has more than 1 placement config so picking first availability domain", *np.Id)
+	}
+
+	// Get the availability domain which is by default in the format of `Uocm:PHX-AD-1`
+	// and remove the hash prefix.
+	availabilityDomain := strings.Split(*np.NodeConfigDetails.PlacementConfigs[0].AvailabilityDomain, ":")[1]
+	return availabilityDomain, nil
+}
+
+func buildGenericLabels(nodePool *oke.NodePool, nodeName string, availabilityDomain string) map[string]string {
+	result := make(map[string]string)
+
+	// Check if this is an ARM node pool and label the nodes appropriately
+	if compiledArmRegexp.MatchString(*nodePool.NodeShape) {
+		result[kubeletapis.LabelArch] = npconsts.ArmArch
```

Comment:
This function is _almost_ identical to `BuildGenericLabelsForInstancePool` in `oci_instance_pool_manager.go` except that the instance pool implementation appears to be missing support for checking for and adding the Ampere / ARM labels if the shape matches. 

Could you add the simple ARM shape check to the instance pool implementation and possibly move the `ArmArch` constant and regular expression into the common package?